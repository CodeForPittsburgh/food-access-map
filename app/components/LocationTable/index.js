/**
 *
 * Location Table
 *
 */

import React, { memo } from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import MaterialTable from 'material-table';

// import PropTypes from 'prop-types';
// import styled from 'styled-components';

/* eslint-disable react/prefer-stateless-function */
class LocationTable extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      locations: [],
    };

    this.loadLocationsFromServer = this.loadLocationsFromServer.bind(this);
  }

  // load location data from the back end, once there is a back end
  loadLocationsFromServer() {
    // axios.get('http://ourBackEnd/api/Locations')
    //   .then(res => {
    //     this.setState({ locations: res.data });
    //   })
    //   .catch(err => {
    //     console.error(err);
    //   });
    const fakeLocations = [
      {
        name: 'dingle',
        open: '8am',
        close: '6pm',
      },
      {
        name: 'dangle',
        open: '7am',
        close: '5pm',
      },
      {
        name: 'dongle',
        open: '6am',
        close: '4pm',
      },
    ];

    this.setState({ locations: fakeLocations });
  }

  componentDidMount() {
    this.loadLocationsFromServer();
  }

  render() {
    return (
      <div>
        <h1>Option 1</h1>
        {/* <p>
          This is one using some more basic material UI components if we want a
          lot of control of everything. Until we know what data we get and what
          we want to be able to do with it, it is hard to say. Could easily add
          custom text input sorting, dropdown sorting for times, etc.
        </p> */}
        <Paper>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Name</TableCell>
                <TableCell align="right">Open</TableCell>
                <TableCell align="right">Close</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {this.state.locations.map(loc => (
                <TableRow key={loc.name}>
                  <TableCell component="th" scope="row">
                    {loc.name}
                  </TableCell>
                  <TableCell align="right">{loc.open}</TableCell>
                  <TableCell align="right">{loc.close}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </Paper>
        <h1>Option 2</h1>
        {/* <p>
          This prebuilt tool looks pretty good. we get filtering out of the box,
          looks easy to change stying, easy to add action buttons for things
          like add/edit, select boxes, individual column filters, custom
          sorting. Only would need above option if we need something REALLY
          custom. Only downside is their dependencies seem brittle (was hard to
          install correctly) and may be tough to keep updated
        </p> */}
        <MaterialTable
          title="Food Banks"
          columns={[
            { title: 'Name', field: 'name' },
            { title: 'Opening Time', field: 'open' },
            { title: 'Closing Time', field: 'close' },
          ]}
          data={this.state.locations}
        />
      </div>
    );
  }
}

// LocationTable.propTypes = {};

export default memo(LocationTable);
