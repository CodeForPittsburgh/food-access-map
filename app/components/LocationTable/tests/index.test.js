/**
 *
 * Tests for LocationTable
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import LocationTable from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

configure({ adapter: new Adapter() });
describe('<LocationTable />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <LocationTable />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  // TODO update this test for better mocking once api function is in place
  it('should load the data from the back end', () => {
    const wrapper = shallow(<LocationTable />);
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'setState');
    instance.loadLocationsFromServer();
    expect(spy).toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <LocationTable />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
