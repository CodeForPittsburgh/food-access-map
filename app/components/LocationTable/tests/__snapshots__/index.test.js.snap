// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationTable /> Should render and match the snapshot 1`] = `
<div>
  <h1>
    Option 1
  </h1>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Open
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Close
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            scope="row"
          >
            dingle
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            8am
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            6pm
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            scope="row"
          >
            dangle
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            7am
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            5pm
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            scope="row"
          >
            dongle
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            6am
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            4pm
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h1>
    Option 2
  </h1>
  <div
    class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    style="position: relative;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MTableToolbar-root-318 MuiToolbar-gutters"
    >
      <div
        class="MTableToolbar-title-322"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Food Banks
        </h6>
      </div>
      <div
        class="MTableToolbar-spacer-320"
      />
      <div
        class="MuiFormControl-root MuiTextField-root MTableToolbar-searchField-323"
        color="inherit"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
              title="Search"
            >
              search
            </span>
          </div>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
                >
                  clear
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="MTableToolbar-actions-321"
      >
        <div>
          <div />
        </div>
      </div>
    </div>
    <div
      style="overflow-x: auto;"
    >
      <div>
        <div
          style="overflow-y: auto;"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MTableHeader-header-426 MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      aria-roledescription="Draggable item. Press space bar to lift"
                      data-react-beautiful-dnd-drag-handle="1"
                      data-react-beautiful-dnd-draggable="1"
                      draggable="false"
                      tabindex="0"
                    >
                      Name
                    </div>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MTableHeader-header-426 MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      aria-roledescription="Draggable item. Press space bar to lift"
                      data-react-beautiful-dnd-drag-handle="1"
                      data-react-beautiful-dnd-draggable="1"
                      draggable="false"
                      tabindex="0"
                    >
                      Opening Time
                    </div>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MTableHeader-header-426 MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      aria-roledescription="Draggable item. Press space bar to lift"
                      data-react-beautiful-dnd-drag-handle="1"
                      data-react-beautiful-dnd-draggable="1"
                      draggable="false"
                      tabindex="0"
                    >
                      Closing Time
                    </div>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                actions=""
                class="MuiTableRow-root"
                index="0"
                level="0"
                localization="[object Object]"
                path="0"
                style="transition: all ease 300ms;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="dingle"
                >
                  dingle
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="8am"
                >
                  8am
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="6pm"
                >
                  6pm
                </td>
              </tr>
              <tr
                actions=""
                class="MuiTableRow-root"
                index="1"
                level="0"
                localization="[object Object]"
                path="1"
                style="transition: all ease 300ms;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="dangle"
                >
                  dangle
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="7am"
                >
                  7am
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="5pm"
                >
                  5pm
                </td>
              </tr>
              <tr
                actions=""
                class="MuiTableRow-root"
                index="2"
                level="0"
                localization="[object Object]"
                path="2"
                style="transition: all ease 300ms;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="dongle"
                >
                  dongle
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="6am"
                >
                  6am
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  value="4pm"
                >
                  4pm
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                style="height: 49px;"
              />
              <tr
                class="MuiTableRow-root"
                style="height: 49px;"
              />
              <tr
                class="MuiTableRow-root"
                style="height: 1px;"
              />
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <table
      class="MuiTable-root"
    >
      <tfoot
        class="MuiTableFooter-root"
        style="display: grid;"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-footer"
        >
          <td
            class="MuiTableCell-root MuiTablePagination-root Component-paginationRoot-287 MuiTableCell-footer"
            colspan="3"
            icons="[object Object]"
            style="float: right; overflow-x: auto;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-288 MuiToolbar-gutters"
            >
              <div
                class="MuiTablePagination-spacer"
              />
              <span
                class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-289 MuiTypography-caption MuiTypography-colorInherit"
              >
                Rows per page:
              </span>
              <div
                class="MuiInputBase-root MuiTablePagination-input"
              >
                <div
                  class="MuiSelect-root MuiTablePagination-selectRoot Component-paginationSelectRoot-290"
                >
                  <div
                    aria-haspopup="true"
                    aria-pressed="false"
                    class="MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      style="padding: 0px 5px;"
                    >
                      5 rows 
                    </div>
                  </div>
                  <input
                    type="hidden"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-289 MuiTypography-caption MuiTypography-colorInherit"
              >
                1-3 of 3
              </span>
              <div
                class="MTablePaginationInner-root-480"
              >
                <span
                  class=""
                  title="First Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        first_page
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class=""
                  title="Previous Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        chevron_left
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                  style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                >
                  1-3 of 3
                </span>
                <span
                  class=""
                  title="Next Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        chevron_right
                      </span>
                    </span>
                  </button>
                </span>
                <span
                  class=""
                  title="Last Page"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        last_page
                      </span>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
